internal/statemachine/state_machine_test.go

```diff
@@ -790,3 +785,229 @@ func TestFailedPostProcessGadgetYaml(t *testing.T) {
 		asserter.AssertErrContains(err, "disallowed for security purposes")
 	})
 }
+
+func TestStateMachine_readMetadataJSON(t *testing.T) {
```

Comment:
Not sure if our previous notation was 'correct' in Go terms, but I like to stay consistent. Could we get this called `TestStateMachineReadMetadataJSON` ?

Reply:
The previous notation was not "correct" :). When naming a test in go the idiomatic way is (AFAIK):

- for an exported function: TestFunctionName
- for an unexported function: Test_functionName
- for a method: TestObjectName_functionName

I find this rather good because there is little ambiguity when reading the test name about what is tested.

I can be consistent in this PR with the current naming convention but I suggest we move to the ideniomatic naming convention sometime.

Reply:
I like consistency, but maybe this is the way to go then! Let's leave it as-is.